server/src/main/java/io/whitefox/core/JsonPredicates.java

```diff
@@ -0,0 +1,385 @@
+package io.whitefox.core;
+
+import org.apache.commons.lang3.tuple.Pair;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+
+class EvalContext {
+
+    public EvalContext(Map<String, String> partitionValues, Map<String, Pair<String, String>> statsValues) {
+        this.partitionValues = partitionValues;
+        this.statsValues = statsValues;
+    }
+
+    Map<String, String> partitionValues;
+    Map<String, Pair<String, String>> statsValues;
+}
+
+class OpDataTypes {
+    static final String BoolType = "bool";
+    static final String IntType = "int";
+    static final String LongType = "long";
+    static final String StringType = "string";
+    static final String DateType = "date";
+    static final String FloatType = "float";
+    static final String DoubleType = "double";
+    static final String TimestampType = "timestamp";
+
+    static List<String> supportedTypes = List.of(BoolType, IntType, LongType, StringType, DateType);
+    static List<String> supportedTypesV2 = Stream.concat(supportedTypes.stream(), Stream.of(FloatType, DoubleType, TimestampType)).collect(Collectors.toList());
```

Comment:
what's v1 and v2 👀?

Reply:
v1 does not support float, double and timestamp from their internal implementation;
we can move maybe to only v2? I dont see references in the API to it

https://github.com/delta-io/delta-sharing/commit/d9ef1d8513ed75b413dd530fa3ba3ec393421be1#diff-bf175d00c670ecba97488c8242dc5c8a9c988d43a413beb4b14cc01bd81df30b